lib/changelog_web/oban_resolver.ex

```diff
@@ -0,0 +1,17 @@
+defmodule ChangelogWeb.ObanResolver do
+  @behaviour Oban.Web.Resolver
+
+  @impl true
+  def resolve_user(conn) do
+    conn.assigns.current_user
+  end
+
+  @impl true
+  def resolve_access(user) do
+    if Changelog.Policies.AdminsOnly.index(user) do
+      :all
+    else
+      nil
+    end
+  end
```

Comment:
@sorentwo this currently raises an error in the failure condition. I didn't see a `:none` action or any way of specifying no access. Did I miss something?

Reply:
There isn't an option to block all access. Preventing any access comes down to authentication rather than authorization, so you'd gate this route behind an admin check or a basic auth plug instead.

Reply:
I'm not sure that's a super useful distinction, especially when the function is called `resolve_access` 😎 

We mount Oban Web in our admin scope, but even that uses more fine-grained access controls than "yes" or "no" for using different parts of it.

I view this as a sub-section of our admin. People with a "host" role can use the admin, but only "admin" role people can access this. In other words: all of these people are authenticated, but only some are authorized to use Oban Web.

Reply:
That's entirely fair. Broader access control hasn't come up often because people are usually preventing access to the dashboard entirely.

We'd like to support it, but haven't had much call for it compared to other features. Maybe now's the time?

Reply:
Up to you! Obviously, you'll have one customer of this work right away. But I know the other stuff you're working on is high value so I can't judge as well as you can.